core/src/main/kotlin/org/sert2521/sertain/motors/MotorController.kt

```diff
@@ -112,155 +44,69 @@ class MotorController<T : MotorId>(
         checkNotNull(encoder) { "You must configure your encoder to use units." }
         setTargetPosition(position.convertTo(encoder!!.ticks).value.toInt())
     }
 
-    fun setTargetVelocity(velocity: Int) {
-        ctreMotorController.set(CtreControlMode.Velocity, velocity.toDouble())
-    }
-
     fun <U1 : AngularUnit, U2 : ChronicUnit, V : AngularVelocityValue<U1, U2>> setTargetVelocity(velocity: V) {
         checkNotNull(encoder) { "You must configure your encoder to use units." }
         setTargetVelocity(velocity.convertTo(encoder!!.ticksPerSecond).value.toInt())
     }
 
-    fun setCurrent(current: Double) {
-        ctreMotorController.set(CtreControlMode.Current, current)
-    }
-
-    fun disable() {
-        ctreMotorController.neutralOutput()
-    }
+    abstract fun setCurrent(current: Double)
 
-    internal fun updatePidf(slot: Int, pidf: MotorPidf) {
-        with(pidf) {
-            ctreMotorController.apply {
-                config_kP(slot, kp)
-                config_kI(slot, ki)
-                config_kD(slot, kd)
-                config_kF(slot, kf)
-                config_IntegralZone(slot, integralZone)
-                configAllowableClosedloopError(slot, allowedError)
-                configMaxIntegralAccumulator(slot, maxIntegral)
-                configClosedLoopPeakOutput(slot, maxOutput)
-                configClosedLoopPeriod(slot, period)
-            }
-        }
-    }
+    abstract fun disable()
 
-    private fun updateCurrentLimit(limit: CurrentLimit) {
-        eachTalon {
-            (ctreMotorController as CtreTalon).apply {
-                configContinuousCurrentLimit(limit.continuousLimit)
-                configPeakCurrentLimit(limit.maxLimit)
-                configPeakCurrentDuration(limit.maxDuration)
-                enableCurrentLimit(limit.enabled)
-            }
-        }
-    }
+    internal abstract fun updatePidf(slot: Int, pidf: MotorPidf)
+}
 
-    init {
-        eachMotor { ctreMotorController.setNeutralMode(ctreNeutralMode(brakeMode)) }
-        ctreMotorController.apply {
-            configClosedloopRamp(closedLoopRamp)
-            configOpenloopRamp(openLoopRamp)
-            configNominalOutputReverse(minOutputRange.start)
-            configNominalOutputForward(minOutputRange.endInclusive)
-            configPeakOutputReverse(maxOutputRange.start)
-            configPeakOutputForward(maxOutputRange.endInclusive)
-            selectProfileSlot(pidfSlot, 0)
-            pidf.toMap().forEach {
-                updatePidf(it.key, it.value)
-            }
-            updateCurrentLimit(currentLimit)
-        }
-        configure()
+fun<T: MotorId> motorController(id: T, configure: MotorController.() -> Unit = {}): MotorController? {
+    return when (id) {
+        is TalonId  -> TalonMotorController(id,  configure = configure)
+        is VictorId -> VictorMotorController(id, configure = configure)
+        else        -> null
```

Comment:
I think you should probably throw some kind of NotImplemented error here, that way the return type can be `MotorController` instead of `MotorController?`